torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -325,6 +325,40 @@ def from_hp_to_floatx(
         else:
             raise NotImplementedError(f"Unsupported dtype {target_dtype} for from_hp_to_floatx")
 
+    @classmethod
+    def from_hp_to_floatx_static(
+        cls,
+        input_float: torch.Tensor,
+        scale: torch.Tensor,
+        block_size: Tuple[int, ...],
+        target_dtype: torch.dtype,
+        layout_type: LayoutType,
+    ):
+
+        if target_dtype in FP8_TYPES:
+            original_shape = input_float.shape
+            input_float = layout_type.pre_process(input=input_float)
+            quant_min, quant_max = math.ceil(torch.finfo(target_dtype).min), math.ceil(torch.finfo(target_dtype).max)
+            zero_point, zero_point_domain = None, None
+
+            int_data = quantize_affine(input_float, block_size, scale, zero_point, target_dtype, quant_min, quant_max, zero_point_domain)
```

Comment:
I thought we want to have a few new quant primitives for float8?

Reply:
We're using the same code from intx, but with a few fixed param values.

Reply:
If thats case couldnt we align the impl similiar to: https://github.com/pytorch/ao/blob/e05635eb25903e8d623c1c88aad44339c41789c3/torchao/dtypes/affine_quantized_tensor.py#L323

where we apply constraints then pass through?

Reply:
zero_point and zero_point_domain are required values for intx_static, while not needed for floatx_static. To make the same call like from_hp_to_floatx, I'll have to update the intx_static definition to accept None for zero_point and zero_point_domain, and then add an assert to check if the target_dtype != float8, then the values can't be None. @jerryzh168 is that okay?

Reply:
yeah that's fine, I agree reusing from_hp_to_intx makes more sense

Reply:
Ok, made those changes.